python/python/lance/dependencies.py

```diff
@@ -0,0 +1,211 @@
+# Originally from https://github.com/pola-rs/polars/blob/ea4389c31b0e87ddf20a85e4c3797b285966edb6/py-polars/polars/dependencies.py
```

Comment:
At what point are modifications substantial enough for a vendored file to receive its own copyright header I wonder?

@changhiskhan

Reply:
i'm not sure. is there a general convention that popular oss projects follow?

Reply:
I was mainly following @wjones127's suggestion [here](https://github.com/lancedb/lance/issues/1217#issuecomment-1785487288).

Reply:
I've updated this section to make it clear the original code was taken from Polars and covered by their license, and subsequent modifications have been made by us which are covered by our license.

I've also added a section to our main license file, similar to how Apache Arrow does it: https://github.com/apache/arrow/blob/main/LICENSE.txt